src/lib/AutocompleteTextInput.svelte

```diff
@@ -0,0 +1,35 @@
+<script lang="ts">
+    import { FormElement } from "./index.js";
+    import { v4 as uuidv4 } from "uuid";
+  
+    export let label: string;
+    export let name: string;
+    export let value: string;
+    export let options: [string,string][];
+    export let dataTestId: string = "";
+    
+    let id = `${name}-${uuidv4()}` 
+    // @ts-expect-error we know the id will be the right thing
+    let dataList: HTMLDataListElement = document.getElementById(`${id}-list`);
+
+    $: {

```

Comment:
See the example https://svelte.dev/repl/2230a1ec6a374c9da13a4e55a1e8701c?version=3.50.1. Svelte can create elements in a much simpler way that also handles destroying elements

Reply:
In retrospect this is obvious, thanks!